test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/utils/ChatContainer.spec.js

```diff
@@ -184,6 +184,31 @@ describe("ChatContainer", () => {
         expect(secondCallToSendDataPass).toMatchObject({ testDataPass: 'worked' });
     });
 
+    it("Mix: process external links, with error returned from JsonUtils bacause of malformed datapass test", () => {
+        chatContainer = new ChatContainer(null, null, mockSDK);
+
+        const externalLinkEventWithInvalidJson = {
+            target : {
+                dataset: {
+                    "nuanceMessageText": "Where do you live?",
+                    "nuanceDatapass": "{'testDataPass': worked'}",//missing ' before worked
+                    "target": "_blank"
+                },
+                getAttribute : jest.fn().mockReturnValue("https://www.gov.uk/government/organisations/hm-revenue-customs")
+            },
+            preventDefault: jest.fn()
+        }
+
+        sanitiseAndParseJsonData = jest.spyOn(JsonUtils, 'sanitiseAndParseJsonData');
+
+        chatContainer.processTranscriptEvent(externalLinkEventWithInvalidJson);
+
+        expect(sanitiseAndParseJsonData).toBeCalledTimes(1);
+        const callToSendDataPass = mockSDK.sendDataPass.mock.calls[1][0];
+
+        expect(callToSendDataPass).toMatchObject({ });
```

Comment:
this highlights something that's probably incorrect in our code - we probably don't want to do the subsequent call if the JSON hasn't been parsed correctly